### DISCUSSION TITLE
native R interface to .cool and .mcool files

### DISCUSSION THREAD

**Discussion by mblanche** [open2c/cooler#374] [Show and tell]
Hi guys, We, at Dovetail Genomics, have written a small interface to the .cool and .mcool files using the HDF5 native R interface. I am planning on making it public and submitting it to BioConductor. I would like to call it coolR but wanted to check with you and Dr. Mirny if you were ok with us using that name. It's an easy mnemonics that will help propagation.

Also, let me know if you would like to have a preview to the interface. Right now it's still a private repos as we are finalizing testing and documentations but I'd be happy to  grant you access. 

Best, 

Marco

**Comment by nvictus** [comment]
Hi Marco,

Wow, that's fantastic! And it's exactly the name I would have recommended, so go for it! Thank you and the folks at Dovetail for developing it and for making it public.

We're not heavy R users over here, as you can probably tell, but I'm happy to give it a try and provide feedback (time allowing).

-Nezar

**Comment by mblanche** [comment]
Nezar, if you want to take a look at the coolR package, install it from this repos: [https://github.com/dovetail-genomics/coolR](https://github.com/dovetail-genomics/coolR). You first need to install the packages InteractionSet and rhdf5 from bioconductor (once it's in BioC, the dependencies will be automatically taken care of):

```
BiocManager::install("InteractionSet")
BiocManager::install("rhdf5")
```
then  I recommend using devtools to install from github:

```
install.packages('devtools')
devtools::install_github("git@github.com:dovetail-genomics/coolR.git")
```

Right now the documentation is sparse, we are working on but I wanted to share it with you in case you think we should add functionalities or modify something.

Let us know,

Best, Marco

**Comment by nvictus** [comment]
I just remembered: I should point you to this earlier conversation on R bindings in case it is useful: https://github.com/mirnylab/cooler/issues/175
